### ISSUE TITLE
Document gradient restrictions for odeint

### ISSUE THREAD

**Issue by john-m-jumper** [google/jax#3557]
`jax.experimental.ode.odeint` takes a function `func(y, t, *args)`.  A known limitation of `odeint` is that `func` cannot close over any tensors, but this information is missing from the documentation.

**Comment by froystig** [comment]
This is also the "documentation" aspect of #2718.

**Comment by mattjj** [comment]
Nice catch, @froystig ! Should we consider this a duplicate, and close one of them (pointing to the other)?

**Comment by froystig** [comment]
This issue could track documentation for the lack of closure support, especially if we think it's easier to close sooner than all of #2718. Then #2718 remains to request support for closures and #3558 depends on how that's done.

**Comment by mattjj** [comment]
#3562 fixed this!
